tests/recipes/test_full_finetune_single_device.py

```diff
@@ -263,3 +264,71 @@ def test_gradient_accumulation(self, tmpdir, monkeypatch):
 
         accum_loss = np.mean(get_loss_values_from_metric_logger(grad_accum_log_file))
         torch.testing.assert_close(no_accum_loss, accum_loss, atol=1e-5, rtol=1e-5)
+
+
+class TestFullFinetuneInt8MixedPrecisionTraining:
+    def _get_test_config_overrides(self):
+        return [
+            "seed=9",
+            "epochs=1",
+            "max_steps_per_epoch=5",
+            "optimizer=torch.optim.AdamW",
+            "optimizer_in_bwd=False",
+            "compile=True",
+        ] + dummy_alpaca_dataset_config()
+
+    @pytest.mark.integration_test
+    def test_smoke(self, tmpdir, monkeypatch):
```

Comment:
couldn't make speed test pass, probably because the model is too small in CI. Anyway, speed benchmarks don't seem to be reliable without a controlled setup. So changed this to smoke test i.e. make sure it works.

Reply:
I believe that its still not passing in the CI because number of ranks is larger than number of gpus available. I think we need this flag `@gpu_test(gpu_count=1)`

https://github.com/pytorch/torchtune/blob/0a3762d058fd9d860606a3a6bbebf71ac593ab5c/tests/recipes/test_full_finetune_distributed.py#L63C5-L63C27

Reply:
Yeah looks like that plus a version check which currently fails on even the GPU runners. Can add something like `@pytest.mark.skipif(not _SUPPORTS_INT8_MIXED_PRECISION_TRAINING)`

Reply:
But agreed on not doing speed test in here, correctness should be the main focus. If you're able to get reproducible loss values that would be a good sanity check. Down the line I'd like to add some performance testing in our currently-very-underutilized `tests/regression_tests` but that's a bigger effort that we shouldn't block on here.